syntax = "proto3";

package TransportGuide;

message RoutingSettings {
  uint32 bus_wait_time = 1;
  double bus_velocity = 2;
}

message SvgPoint {
  double x = 1;
  double y = 2;
}

message Rgb {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
}

message Rgba {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
  double alpha = 4;
}

message Color {
  oneof color_definition {
    Rgb rgb = 1;
    Rgba rgba = 2;
    string name = 3;
  }
}

message RenderSettings {
  double width = 1;
  double height = 2;
  double padding = 3;
  double stop_radius = 4;
  double line_width = 5;
  int32 stop_label_font_size = 6;
  SvgPoint stop_label_offset = 7;
  Color underlayer_color = 8;
  double underlayer_width = 9;
  repeated Color color_palette = 10;
  int32 bus_label_font_size = 11;
  SvgPoint bus_label_offset = 12;
  repeated string layers = 13;
  double outer_margin = 14;
};

message RouteInfo {
  uint64 id = 1;
  double weight = 2;
  uint64 edge_count = 3;
  repeated uint64 expanded_route = 4;
  string vertex_from = 5;
  string vertex_to = 6;
};

message WaitActivity {
  string stop_name = 1;
  SvgPoint map_point = 2;
}

message BusActivity {
  double time = 1;
  string bus = 2;
  uint32 span_count = 3;
  uint32 start_stop_idx = 4;
}

message Router {
  RoutingSettings settings = 1;
  RenderSettings render_settings = 2;
  repeated RouteInfo route_info = 3;
  repeated WaitActivity wait_activity = 4;
  repeated BusActivity bus_activity = 5;
  string base_map = 6;
}
